IpmiFeaturePkg/GenericIpmi/Common/GenericIpmi.c

```diff
@@ -277,7 +277,7 @@ IpmiSendCommandInternal (
     //
     // Verify the response data buffer passed in is big enough.
     //
-    if ((DataSize - IPMI_RESPONSE_HEADER_SIZE) > *ResponseDataSize) {
+    if ((ResponseDataSize != NULL) && (DataSize - IPMI_RESPONSE_HEADER_SIZE) > *ResponseDataSize){

```

Comment:
I would have a check at the beginning that will return invalid parameter if `(ResponseDataSize == NULL) != (ResponseData)`

Then you can just put this whole block under a `(ResponseData != NULL)` check instead of two different checks.

Reply:
Sure, but my point is that a caller should never pass NULL for one of those but not the other. Adding that check before doing any work would have clarify the behavior of this functionas well as allow you to clean up and centralize these checks. specifically,  `(ResponseDataSize == NULL) != (ResponseData)`

Reply:
Hi Chris,

I have made changes in function header routine in both c & .h files as below:

  IN OUT  UINT8           *ResponseData OPTIONAL,
  IN OUT  UINT8           *ResponseDataSize OPTIONAL

For your comment for the check, still I am unsure why it required to be at the early and a centralised check.

Even though these 2 passed arguments are a pointer of a data and its size, they both are 2 different entities passed as pointer for the routine. Being an optional parameter, the check is being made only when is being used and is not interdependent on each other.  That is the way code has been modified.

By this way it takes care of  cases: Both parameters are NULL,  Both parameters are NOT NULL and one of the parameter is NULL. 

Early check and breaking with a return would still will not execute the required code in NULL passed with intentionally required,

Do clarify me on if I am not missing any case that you are foreseeing.